crates/lib3h/src/transport/error.rs

```diff
@@ -10,98 +10,144 @@ pub type TransportResult<T> = Result<T, TransportError>;
 
 /// represents an error generated by a connection instance
 #[derive(Debug, PartialEq, Clone)]
-pub struct TransportError(pub String);
+pub struct TransportError(Box<ErrorKind>);
```

Comment:
Surprsied to see a boxed type here- not a huge deal, but what precipitated it?

Reply:
I was just copying how this was done in our other error types.  Do you think we should remove it everywhere?

Reply:
For this PR just leave it alone.